version: "3"
services:
  postgres:
    image: library/postgres:13
    container_name: postgres
    restart: on-failure
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: "postgres"
      POSTGRES_DB: postgres
    volumes:
      - db-data:/var/lib/postgresql/data
      - ./postgres/:/docker-entrypoint-initdb.d/
    ports:
      - "5432:5432"
    networks:
      - database
    healthcheck:
      test:
        - "CMD-SHELL"
        - "pg_isready -d $$POSTGRES_DB -U $$POSTGRES_USER"
      interval: 5s
      timeout: 5s
      retries: 5

  person-service:
    image: persons-apiserver:${TAG}
    build: .
    container_name: person-service
    ports:
      - "8080:8080"
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - ./src/config-gh.yaml:/usr/local/etc/apiserver/config.yaml
    networks:
      - api-gateaway
      - database
    healthcheck:
      test: ["CMD", "wget", "--no-verbose", "--tries=1", "--spider", "http://person-service:8080/readiness"]
      interval: 10s
      timeout: 5s
      retries: 6
      start_period: 10s

volumes:
  db-data:

networks:
  api-gateaway:
    driver: bridge

  database:
    driver: bridge
    internal: true
